sympy/combinatorics/homomorphisms.py

```diff
@@ -533,8 +535,7 @@ def find_homomorphism(G, H, injective=True, surjective=True, compute=True, all=F
             if isinstance(H, FpGroup):
                 images = h_isomorphism.invert(images)
             T =  homomorphism(G, H, G.generators, images, check=False)
-            if ((injective and surjective and T.is_isomorphism()) or
-                    (injective and T.is_injective()) or (surjective and T.is_surjective())):
+            if (injective == T.is_injective()) or (surjective == T.is_surjective()):
```

Comment:
Shouldn't this be `and`?

Reply:
Well, if we use an `and` we might miss out the cases where it is both epimorphism/monomorphism and also and isomorphism whenever only one of `injective`or `surjective` is `True`, right?

Reply:
An isomorphism is an epimorphism (and also a monomorphism). Using `or` would actually make us find more homomorphisms than needed: for example, if `injective=True`, this condition will be true for any injective homomorphism, surjective or not, - and they'd all be returned, even if the user specified `surjective=True`